types/jest/index.d.ts

```diff
@@ -805,8 +857,11 @@ declare namespace jest {
          * Use to test the specific value that a mock function returned for the nth call.
          * If the nth call to the mock function threw an error, then this matcher will fail
          * no matter what value you provided as the expected return value.
+         *
+         * Optionally, you can provide a type for the expected value via a generic.
+         * This is particuarly useful for ensuring expected objects have the right structure.
          */
-        toHaveNthReturnedWith(nthCall: number, expected: any): R;
+        toHaveNthReturnedWith<E>(nthCall: number, expected: E): R;
```

Comment:
```suggestion
        toHaveNthReturnedWith<E = any>(nthCall: number, expected: E): R;
```

Explicit fallback rather than implicit